### DISCUSSION TITLE
Caculation of Stress in GNUPLOT script in tensile test example !!!

### DISCUSSION THREAD

**Discussion by ppandit95** [peridigm/peridigm#214] [General]
Dear Peridigm Community

As I was working on implementing a text file geometry with the tensile test example, I was wondering about a factor of 0.5 used in plotgen script as can be seen below 

 cross section of specimen = (0.124 in)(0.25 in) = 0.031 in^2 = 0.1999996 cm^2
 1 dyne/cm^2 = 1.0e-10 GPa

plot simulation results against elastic modulus
E = 180.0
Modulus(x) = E * x

set terminal pdf enhanced font "Times-Roman,24" size 12in, 8in 
set output "tensile_test.pdf"
set xlabel "Engineering Strain (m/m)" font "Times-Roman,32"
set ylabel "Engineering Stress (GPa)" font "Times-Roman,32"
set key bottom right
plot "1BLOCK_OUTPUT.csv" using ($4-$5)/($2-$3):($7-$6)  * 0.5 * 1.0e-10/0.1999992 with points pt 7 ps 4 lc 3 title "Peridynamic Simulation", \
     Modulus(x) with lines lw 10 title "Elastic Modulus"

As I couldn't make sense of it and since I wasn't getting the desired elastic moduli with the new geometry I was wondering what could be the meaning of this factor "0.5".

Many Thanks
Pushkar

**Comment by djlittl** [comment]
I’m pretty sure the factor of 0.5 is there because the reaction forces at the top and bottom of the specimen are being summed together and then divided by 2, to get an average value.  The factor of 0.5 is the division by 2.  This step shouldn’t technically be required, one should be able to just use one of the reaction forces because they should be equal and opposite.  But I suppose they won’t be exactly equal and opposite due to solver tolerances and whatnot.

From: Pushkar Anirudha Pandit ***@***.***>
Reply-To: peridigm/peridigm ***@***.***>
Date: Thursday, February 10, 2022 at 5:03 AM
To: peridigm/peridigm ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] [peridigm/peridigm] Caculation of Stress in GNUPLOT script in tensile test example !!! (Discussion #214)


Dear Peridigm Community

As I was working on implementing a text file geometry with the tensile test example, I was wondering about a factor of 0.5 used in plotgen script as can be seen below

cross section of specimen = (0.124 in)(0.25 in) = 0.031 in^2 = 0.1999996 cm^2
1 dyne/cm^2 = 1.0e-10 GPa
plot simulation results against elastic modulus

E = 180.0
Modulus(x) = E * x

set terminal pdf enhanced font "Times-Roman,24" size 12in, 8in
set output "tensile_test.pdf"
set xlabel "Engineering Strain (m/m)" font "Times-Roman,32"
set ylabel "Engineering Stress (GPa)" font "Times-Roman,32"
set key bottom right
plot "1BLOCK_OUTPUT.csv" using ($4-$5)/($2-$3):($7-$6)0.51.0e-10/0.1999992 with points pt 7 ps 4 lc 3 title "Peridynamic Simulation",
Modulus(x) with lines lw 10 title "Elastic Modulus"

As I couldn't make sense of it and since I wasn't getting the desired elastic moduli with the new geometry I was wondering what could be the meaning of this factor "0.5".

Many Thanks
Pushkar

—
Reply to this email directly, view it on GitHub<https://github.com/peridigm/peridigm/discussions/214>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABKXABITXYJRO6JQJXEUPTTU2OSSFANCNFSM5OAPWOFQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by ppandit95** [reply]
Thanks @djlittl for elaborating on it but for my new geometry I did get the desired peridynamic modulus equal to elastic modulus with a factor of 0.82 I guess I need to verify this by running the simulation for a larger time and checking if I still get the desired elastic modulus.

**Comment by djlittl** [comment]
Does your updated geometry have a different cross-sectional area than the original geometry?  For the plot, engineering stress is computed by dividing the total reaction force on one of the ends of the bar by the cross-sectional area.  I think the cross-sectional area is hard coded in the gnuplot script.  This could explain why you need to tweak the multiplier.

Changing the multiplier without understanding why it needs to be changed strikes me as very dubious.  If you’re using a correspondence material model, the simulation should give you a very accurate value for the modulus.  If you’re using a different type of material model, the simulation should give you a pretty good match to the modulus, and should become more accurate as you reduce the horizon (which would require you to refine the mesh) so that you reduce the surface effect.

From: Pushkar Anirudha Pandit ***@***.***>
Reply-To: peridigm/peridigm ***@***.***>
Date: Thursday, February 10, 2022 at 9:30 PM
To: peridigm/peridigm ***@***.***>
Cc: "Littlewood, David John" ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [peridigm/peridigm] Caculation of Stress in GNUPLOT script in tensile test example !!! (Discussion #214)


Thanks @djlittl<https://github.com/djlittl> for elaborating on it but for my new geometry I did get the desired peridynamic modulus equal to elastic modulus with a factor of 0.82 I guess I need to verify this by running the simulation for a larger time and checking if I still get the desired elastic modulus.

—
Reply to this email directly, view it on GitHub<https://github.com/peridigm/peridigm/discussions/214#discussioncomment-2154831>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABKXABKR7XW5SPRH4ZZBQRTU2SGHDANCNFSM5OAPWOFQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by ppandit95** [reply]
Thanks @djlittl for suggesting that and I have used a geometry which is quiet similar to original tensile test specimen but instead of a dog bone specimen I have a simple rectangular bar which has twice the area of our original tensile test specimen (i.e. twice the cross section area of testing portion of the entire dog bone geometry) and have aptly modified the gnuplot script but still I require modifications in multiplier which I feel is due to change in surface effects due to little modification in geomerty.So I wanted to know whether I am missing something here ?

**Comment by djlittl** [comment]
Adding an extra multiplier doesn't sound right.  There shouldn't be a need for that, especially if you're using a correspondence model, which doesn't have much of a surface effect.  I suggest you try to recover the modulus with a simple calculation outside the gnuplot script, to make sure you get the expected result.  Engineering strain from the (virtual) strain gauge, and engineering stress by dividing the reaction force by the cross-sectional area.

One other thought that comes to mind is to make sure you're properly capturing the reaction force.  There are blocks set up in the geometry that are used to compute the reaction force by summing all the net forces on all the nodes in that block.  If you changed the horizon or the size of these blocks, then you could be accidentally excluding some nodes that have nonzero net forces from the calculation of reaction force.

**Reply by ppandit95** [reply]
that was indeed insightful @djlittl and I will try to recover the elastic moduli but I haven't changed the number of material points on the boundary blocks to compute forces but still I will look into this as I initially didn't knew that correspondence model has less surface effects
